plugins/woocommerce-blocks/tests/e2e/tests/checkout/checkout-block.shopper.block_theme.side_effects.spec.ts

```diff
@@ -598,5 +598,26 @@ test.describe( 'Billing Address Form', () => {
 				}
 			}
 		} );
+
+		test( 'Ensure that each <CheckboxControl> component has a unique id', async ( {
+			frontendUtils,
+			page,
+		} ) => {
+			await frontendUtils.emptyCart();
+			await frontendUtils.goToShop();
+			await frontendUtils.addToCart( SIMPLE_PHYSICAL_PRODUCT_NAME );
+			await frontendUtils.goToCheckout();
+
+			const isUnique = await page.evaluate( () => {
+				const elements = document.querySelectorAll(
+					'[id^="checkbox-control-"]'
+				);
+				const ids = Array.from( elements ).map( ( el ) => el.id );
+				const uniqueIds = new Set( ids );
+				return ids.length === uniqueIds.size;
+			} );
+
+			expect( isUnique ).toBe( true );
```

Comment:
This implementation may not be testing what's intended because we're not asserting an explicit number of checkboxes to be validated, so we cannot be sure that we're validating all the checkboxes at the moment of query. Once we know how many checkboxes there are (which is a must to make this test deterministic), we could use the API that utilizes the auto-waiting mechanism (`toHaveCount`) to stabilize this test. For example:

```js
const checkboxes = page.locator( '[id^="checkbox-control-"]' );

await expect( checkboxes ).toHaveCount( X );

const ids = checkboxes.evaluateAll( ( nodes ) =>
	nodes.map( ( node ) => node.id )
);
const uniqueIds = new Set( ids );

expect( ids.length ).toEqual( uniqueIds.size );
```

Reply:
Thanks for your review, @WunderBart. I've adjusted the code based on your feedback and the call with you.